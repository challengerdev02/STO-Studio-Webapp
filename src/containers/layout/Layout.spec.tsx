import { cleanup, render } from '@testing-library/react';
import { MainLayoutContainer, AuthLayoutContainer } from './index';

const mockHandleUpdateAccountFn = jest.fn();
const mockFindOneAccountFn = jest.fn();
const mockPush = jest.fn();

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'), // use actual for all non-hook parts
  useParams: () => ({
    teamId: 'someTeamId',
    eventId: 'someEventId',
    broadcastId: 'someShowId',
  }),
  useRouter: () => ({
    query: {},
    push: mockPush,
    replace: mockPush,
  }),
  useLocation: jest.fn(),
}));

jest.mock('@/hooks', () => {
  return {
    ...jest.requireActual('@/hooks'), // use actual for all non-hook parts

    useBook: () => ({
      book: {},
      handleGetBook: jest.fn(),
    }),
    useAccount: jest.fn().mockImplementation(() => ({
      user: {
        username: 'mike',
        walletAddress: '0xrrr',
        url: 'url string',
        emailAccount: {
          email: 'hs@gimal.vom',
        },
      },
      search: jest.fn(),
      handleUpdateAccount: mockHandleUpdateAccountFn,
      handleFindOneAccount: mockFindOneAccountFn,
    })),
    useUIState: () => ({
      uiLoaders: [],
    }),
  };
});

describe('View MainLayoutContainer Container', function () {
  afterEach(cleanup);
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it('should match MainLayoutContainer with snapshot', () => {
    // @ts-ignore
    const { container } = render(
      <MainLayoutContainer>
        <h1>Hello</h1>
      </MainLayoutContainer>
    );
    expect(container).toMatchSnapshot();
  });

  it('should match AuthLayoutContainer with snapshot', () => {
    // @ts-ignore
    const { container } = render(
      <AuthLayoutContainer>
        <h1>Hello</h1>
      </AuthLayoutContainer>
    );
    expect(container).toMatchSnapshot();
  });
});
