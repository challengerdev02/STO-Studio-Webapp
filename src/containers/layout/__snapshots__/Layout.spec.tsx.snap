// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View MainLayoutContainer Container should match AuthLayoutContainer with snapshot 1`] = `
<div>
  <h1>
    Hello
  </h1>
</div>
`;

exports[`View MainLayoutContainer Container should match MainLayoutContainer with snapshot 1`] = `
<div>
  <main
    class="sc-hmjpVf bbmNDg"
  >
    <header
      class="sc-bTfYFJ jmubKv"
      id="main-header"
      role="main-header"
    >
      <a
        href="/"
      >
        <div
          class="sc-kHOZwM gDIxfl"
        >
          <img
            alt="Metacomics"
            class="sc-hOGkXu ljQWcv md-logo"
            src="/logo.png"
          />
          <img
            alt="Metacomics"
            class="sc-hOGkXu ljQWcv sm-logo"
            src="/icon.png"
          />
        </div>
      </a>
      <div
        class="ant-divider ant-divider-vertical"
        role="separator"
      />
      <div
        class="link-list"
      >
        <a
          class="meta-text-link"
          href="/explore-collections"
        >
          Explore
        </a>
        <a
          class="meta-text-link"
          href="/resources"
        >
          Resources
        </a>
        <a
          class="meta-text-link"
          href="/token"
        >
          Token
        </a>
        <a
          class="meta-text-link"
          href="/launch"
        >
          Launch
        </a>
      </div>
      <div
        class="button-list button-list-desktop"
      >
        <form
          class="ant-form ant-form-horizontal"
          id="search-form"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      class="ant-input"
                      id="search-form_search"
                      placeholder="Item, Collection, Artist"
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <button
                        class="ant-btn ant-btn-text"
                        type="button"
                      >
                        <i
                          class="mc-search-line"
                        />
                      </button>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div>
          <button
            class="ant-btn ant-btn-round ant-btn-primary"
            data-testid="create-button"
            type="button"
          >
            <span>
              Create
            </span>
          </button>
        </div>
        <div
          class="sc-jUosCB fDMkgp"
        >
          <button
            class="ant-btn ant-btn-round ant-btn-default"
            data-testid="wallet-address-button"
            type="button"
          >
            <span>
              0xrrr
            </span>
          </button>
        </div>
        <div
          data-testid="drawer-balance-icon"
          style="cursor: pointer;"
        >
          <span
            class="ant-avatar ant-avatar-circle ant-avatar-icon"
            style="width: 44px; height: 44px; line-height: 44px; font-size: 22px;"
          >
            <svg
              height="44"
              version="1.1"
              viewBox="0 0 80 80"
              width="44"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs>
                <lineargradient
                  id="0xrrr"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    stop-color="rgb(0, 255, 30)"
                  />
                  <stop
                    offset="100%"
                    stop-color="rgb(30, 0, 255)"
                  />
                </lineargradient>
              </defs>
              <g
                fill="none"
                fill-rule="evenodd"
                stroke="none"
                stroke-width="1"
              >
                <rect
                  fill="url(#0xrrr)"
                  height="80"
                  id="Rectangle"
                  width="80"
                  x="0"
                  y="0"
                />
              </g>
            </svg>
          </span>
        </div>
      </div>
      <div
        class="ant-space ant-space-horizontal ant-space-align-center button-list-mobile mobile-list-desktop"
      >
        <div
          class="ant-space-item"
          style="margin-right: 15px;"
        >
          <div>
            <button
              class="ant-btn ant-btn-circle ant-btn-default ant-btn-sm ant-btn-icon-only toggle-menu"
              type="button"
            >
              <i
                class="mc-search-line"
              />
            </button>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-right: 15px;"
        >
          <div>
            <button
              class="ant-btn ant-btn-circle ant-btn-default ant-btn-sm ant-btn-icon-only toggle-menu"
              type="button"
            >
              <i
                class="mc-burger-line"
              />
            </button>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            data-testid="drawer-balance-icon-2"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-icon"
              style="width: 44px; height: 44px; line-height: 44px; font-size: 22px;"
            >
              <svg
                height="44"
                version="1.1"
                viewBox="0 0 80 80"
                width="44"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <defs>
                  <lineargradient
                    id="0xrrr"
                    x1="0%"
                    x2="100%"
                    y1="0%"
                    y2="100%"
                  >
                    <stop
                      offset="0%"
                      stop-color="rgb(0, 255, 30)"
                    />
                    <stop
                      offset="100%"
                      stop-color="rgb(30, 0, 255)"
                    />
                  </lineargradient>
                </defs>
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <rect
                    fill="url(#0xrrr)"
                    height="80"
                    id="Rectangle"
                    width="80"
                    x="0"
                    y="0"
                  />
                </g>
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="mobile-menu-layout"
        style="display: none;"
      >
        <div
          class="logo-container"
        >
          <img
            alt="logo"
            class="sc-hOGkXu ljQWcv md-logo"
            src="/logo.png"
          />
        </div>
        <form
          class="ant-form ant-form-horizontal"
          id="search-form"
        >
          <div
            class="ant-row ant-form-item"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <span
                    class="ant-input-affix-wrapper"
                  >
                    <input
                      class="ant-input"
                      id="search-form_search"
                      placeholder="Item, Collection, Artist"
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <button
                        class="ant-btn ant-btn-text"
                        type="button"
                      >
                        <i
                          class="mc-search-line"
                        />
                      </button>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="links"
        >
          <span
            class="ant-typography"
          >
            Explore
          </span>
          <span
            class="ant-typography"
          >
            Resources
          </span>
          <span
            class="ant-typography"
          >
            Token
          </span>
          <span
            class="ant-typography"
          >
            Launch
          </span>
          <span
            class="ant-typography"
          >
            Create
          </span>
        </div>
      </div>
    </header>
    <section
      class="sc-eLwHnm kJDKTz"
    >
      <h1>
        Hello
      </h1>
    </section>
    <footer
      class="sc-ehCJOs fmdQBk"
    >
      <div
        class="sc-jIkXHa kCNXSw"
      >
        <div
          class="sc-ZOtfp ftztIt"
        >
          <div
            class="sc-TBWPX fSUpLG"
          >
            <img
              alt="logo"
              class="sc-lcepkR kbgJKg"
              src="/test-file-stub"
            />
            <div
              class="sc-jeraig efePAT"
            >
              MetaComic
            </div>
          </div>
          <div
            class="sc-gGCDDS gfesHT"
          >
            For Creators
          </div>
        </div>
        <div
          class="ant-divider ant-divider-horizontal sc-nVkyK KSPNa"
          role="separator"
        />
        <div
          class="sc-hiwPVj kOZoec"
        >
          MetaComic
          <div
            class="sc-clIzBv fDxnhp"
          >
            <div
              class="sc-Galmp eEwjGe"
            >
              Explore
            </div>
            <div
              class="sc-Galmp eEwjGe"
            >
              Connect wallet
            </div>
            <div
              class="sc-Galmp eEwjGe"
            >
              Create item
            </div>
          </div>
        </div>
        <div
          class="ant-divider ant-divider-horizontal sc-nVkyK KSPNa"
          role="separator"
        />
        <div
          class="sc-hiwPVj kOZoec last"
        >
          Community
          <div
            class="sc-clIzBv fDxnhp"
          >
            <div
              class="sc-Galmp eEwjGe"
            >
              MetaComic Token
            </div>
            <div
              class="sc-Galmp eEwjGe"
            >
              Discussion
            </div>
            <div
              class="sc-Galmp eEwjGe"
            >
              Voting
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal sc-eJwWfJ cwtoQZ"
        role="separator"
      />
      <div
        class="sc-faUpoM jOnspz"
      >
        <div
          class="sc-fWCJzd eZTfKJ"
        >
          Copyright © 2022 MetaComic. All rights reserved
        </div>
        <div
          class="sc-jOxtWs ifPxJr"
        >
          <div
            class="sc-dvQaRk hAHlPL"
          >
            We use cookies for better service.
            <span>
              <button
                class="ant-btn ant-btn-link"
                style="padding-right: 0px;"
                type="button"
              >
                <span>
                  Accept
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </footer>
  </main>
</div>
`;
